interpreter/src/main/scala/pl/touk/nussknacker/engine/spel/SpelExpressionSuggester.scala

```diff
@@ -0,0 +1,154 @@
+package pl.touk.nussknacker.engine.spel
+
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.generic.JsonCodec
+import org.springframework.expression.spel.SpelNode
+import org.springframework.expression.spel.ast.{Indexer, PropertyOrFieldReference, StringLiteral, VariableReference}
+import pl.touk.nussknacker.engine.TypeDefinitionSet
+import pl.touk.nussknacker.engine.api.context.ValidationContext
+import pl.touk.nussknacker.engine.api.dict.DictQueryService
+import pl.touk.nussknacker.engine.api.typed.typing.{TypedClass, TypedDict, TypedObjectTypingResult, TypedObjectWithValue, TypedUnion, TypingResult}
+import pl.touk.nussknacker.engine.definition.ProcessDefinitionExtractor.ExpressionDefinition
+import pl.touk.nussknacker.engine.definition.TypeInfos.ClazzDefinition
+import pl.touk.nussknacker.engine.dict.SimpleDictRegistry
+import pl.touk.nussknacker.engine.graph.expression.Expression
+import pl.touk.nussknacker.engine.spel.Typer.TypingResultWithContext
+import pl.touk.nussknacker.engine.spel.ast.SpelAst.SpelNodeId
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Try}
+
+class SpelExpressionSuggester(expressionConfig: ExpressionDefinition[_], typeDefinitions: TypeDefinitionSet, dictQueryService: DictQueryService, classLoader: ClassLoader) {
+  private val successfulNil = Future.successful[List[ExpressionSuggestion]](Nil)
+  private val typer = Typer.default(classLoader, expressionConfig, new SimpleDictRegistry(expressionConfig.dictionaries), typeDefinitions)
```

Comment:
We can't just create `SimpleDictRegistry` here. We should pass `UiDictServices` which has both `DictQueryService` and `DictRegistry` implementation from `ModelData`. `SimpleDictRegistry` can't handle other implementations of `DictDefinition` than `EmbeddedDictDefinition`.

Reply:
fixed